subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/ReportFriendlyFailureHandler.kt

```diff
@@ -0,0 +1,185 @@
+            Regex("with decor view of type .+ did not match any of the following roots:"),
+            "Не найдена view в иерархии указанного window"
+        ),
+        RegexToPatternMessageNormalizer(
+            Regex("Found 0 items matching holder with view.+"),
+            "Не найдена view в recycler "
+        ),
+        RegexToPatternMessageNormalizer(
+            Regex("Parameter specified as non-null is null.+parameter (.+)"),
+            "Параметр {1} == null"
+        ),
+        RegexToPatternMessageNormalizer(
+            Regex(".+Perform action single click on.+: \\((.+)\\).+"),
+            "Не удалось кликнуть по элементу в Recycler: {1}"
+        )
+    )
+
+    override fun handle(error: Throwable, viewMatcher: Matcher<View>?) {
+        try {
+            defaultHandler.handle(error, viewMatcher)
+        } catch (e: Throwable) {
+            when {
+                e.isCausedBy<AppNotIdleException> { it.message.orEmpty().contains("ASYNC_TASKS_HAVE_IDLED") } -> {
+                    val asyncTaskThreads = Thread.getAllStackTraces()
+                        .filterKeys { it.name.contains("AsyncTask") }
+                    val threadDump = asyncTaskThreads.entries.joinToString(separator = "\n") { it.dumpState() }
+                    throw createException<AppNotIdleException>(
+                        "AsyncTask is still running in background. Thread dump:\n$threadDump"
+                    )
+                }
+                e.isCausedBy<AppNotIdleException> { it.message.orEmpty().contains("Looped for ") } -> {
+                    throw createException<AppNotIdleException>(
+                        "Main thread is busy. " +
+                            "The probable cause is in animations. " +
+                            "Try to find them and suppress by @SuppressAnimationRule"
+                    )
+                }
+                // Handle expected errors from UITestConfig.waiterAllowedExceptions
+                // TODO: make this contract explicit
+                e is PerformException -> {
+
+                    // RecyclerView descendant checks implemented via ViewActions (gross)
+                    if (e.actionDescription.startsWith("Check descendant view")) {
+                        throw AssertionFailedError("Не прошла проверка: ${e.actionDescription}")
+                    }
+
+                    // rethrow exception with bare minimum of useful info
+                    throw PerformException.Builder()
+                        .from(e)
+                        .withViewDescription(minimizeViewDescription(e.viewDescription))
+                        .withCause(e)
+                        .build()
+                }
+                e is AssertionFailedError -> {
+                    throw AssertionFailedError(e.normalizedMessage())
+                }
+                e is NoMatchingRootException -> {
+                    throw createException<NoMatchingRootException>(e.normalizedMessage())
```

Comment:
The difference is here. 
Before we've wrapped it in `UITestFrameworkException`. Now we throw the save Espresso exception.

By default, retry skip these errors:

```kotlin
var waiterAllowedExceptions = setOf(
        EspressoException::class.java,
        AssertionError::class.java
)
```